packages/design-carousel/src/index.tsx

```diff
@@ -1,45 +1,121 @@
+export default function DesignCarousel( {
+	onPick,
+	selectedDesignSlugs,
+	flow,
+}: DesignCarouselProps ) {
 	const { __ } = useI18n();
-
 	const swiperInstance = useRef< Swiper | null >( null );
-
 	const locale = useLocale();
+	const isLargerThan1440px = useMediaQuery( '(min-width: 1440px)' );
+	const mobileOptions: MShotsOptions = { w: 400, vpw: 400, vph: 872, format: 'png' };
+	const desktopOptions: MShotsOptions = {
+		w: isLargerThan1440px ? 1920 : 1280,
+		vpw: 1920,
+		vph: 1280,
+		format: 'png',
+	};
 
 	const { data: allDesigns } = useStarterDesignsQuery( {
 		_locale: locale,
 	} );
 
-	let selectedDesigns = allDesigns?.designs;
+	const shouldDisplayMobileViewOnly = ( flow: string | undefined ) => {
+		switch ( flow ) {
+			case 'link-in-bio':
+				return true;
+			default:
+				return false;
+		}
+	};
 
-	if ( selectedDesigns && selectedDesignSlugs ) {
-		// If we have a restricted set of designs, filter out all unwanted designs
-		const filteredDesigns = selectedDesigns.filter( ( design ) =>
-			selectedDesignSlugs.includes( design.slug )
-		);
+	const getLinkInBioDesigns = ( allDesigns: StarterDesigns | undefined ) => {
+		const designs = allDesigns
+			? allDesigns?.designs.filter(
+					( design ) =>
+						design.is_virtual &&
+						design.categories.some( ( category ) => category.slug === 'link-in-bio' )
+			  )
+			: [];
 
-		// Now order the filtered set based on the supplied slugs.
-		selectedDesigns = selectedDesignSlugs
-			.map( ( selectedDesignSlug ) =>
-				filteredDesigns.find( ( design ) => design.slug === selectedDesignSlug )
-			)
-			.filter( ( selectedDesign ) => !! selectedDesign ) as Design[];
-	}
+		return designs;
+	};
+
+	const getEcommerceDesigns = (
+		allDesigns: StarterDesigns | undefined,
+		selectedDesignSlugs: string[] | undefined
+	) => {
+		let selectedDesigns = allDesigns?.designs;
+
+		if ( selectedDesigns && selectedDesignSlugs ) {
+			// If we have a restricted set of designs, filter out all unwanted designs
+			const filteredDesigns = selectedDesigns.filter( ( design ) =>
+				selectedDesignSlugs.includes( design.slug )
+			);
+
+			// Now order the filtered set based on the supplied slugs.
+			selectedDesigns = selectedDesignSlugs
+				.map( ( selectedDesignSlug ) =>
+					filteredDesigns.find( ( design ) => design.slug === selectedDesignSlug )
+				)
+				.filter( ( selectedDesign ) => !! selectedDesign ) as Design[];
+		}
+
+		return selectedDesigns;
+	};
+
+	const getCarouselOptions = ( flow: string | undefined ) => {
+		switch ( flow ) {
+			case 'link-in-bio':
+				return mobileOptions;
+			default:
+				return desktopOptions;
+		}
+	};
+
+	const getFlowDesigns = (
+		allDesigns: StarterDesigns | undefined,
+		locale: string | undefined,
+		selectedDesignSlugs: string[] | undefined,
+		flow: string | undefined
+	) => {
+		if ( ! allDesigns || ! locale ) {
+			return null;
+		}
+
+		switch ( flow ) {
+			case 'link-in-bio':
```

Comment:
Same as above, this may break for other variations of LiB flow (eg `link-in-bio-tld`).

Reply:
Good catch. I updated these cases to use the `isLinkInBioFlow` function.